### ISSUE TITLE
Tests fail: AttributeError: module 'numpy' has no attribute 'bool'

### ISSUE THREAD

**Issue by yurivict** [jonathf/numpoly#119]
```
____________________________________________________________________________ ERROR collecting test/test_align.py ____________________________________________________________________________
test_align.py:6: in <module>
    X, Y = numpoly.variable(2)
../../stage/usr/local/lib/python3.11/site-packages/numpoly/construct/variable.py:44: in variable
    return numpoly.symbols(
../../stage/usr/local/lib/python3.11/site-packages/numpoly/construct/symbols.py:93: in symbols
    out = numpoly.ndpoly.from_attributes(
../../stage/usr/local/lib/python3.11/site-packages/numpoly/baseclass.py:388: in from_attributes
    return numpoly.polynomial_from_attributes(
../../stage/usr/local/lib/python3.11/site-packages/numpoly/construct/from_attributes.py:95: in polynomial_from_attributes
    numpoly.cfrom_attributes(coefficients, poly.values.ravel())
numpoly/cfunctions/cfrom_attributes.pyx:22: in numpoly.cfunctions.cfrom_attributes.cfrom_attributes
    ???
numpoly/cfunctions/cvalues.pyx:124: in numpoly.cfunctions.cvalues.cset_values
    ???
numpoly/cfunctions/cvalues.pyx:129: in numpoly.cfunctions.cvalues.cset_values
    ???
/usr/local/lib/python3.11/site-packages/numpy/__init__.py:324: in __getattr__
    raise AttributeError(__former_attrs__[attr])
E   AttributeError: module 'numpy' has no attribute 'bool'.
E   `np.bool` was a deprecated alias for the builtin `bool`. To avoid this error in existing code, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
E   The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
E       https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
```

Version: 1.3.3
numpoly-1.3.3
Python-3.11
FreeBSD 14.1

**Comment by jonathf** [comment]
What version of numpy?

**Comment by penguinpee** [comment]
I just ran into the same with NumPy 1.26.4.

`np.bool` has been deprecated since 1.20:

https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations

I did replace the two occurrences of `np.bool` with `np.bool_`. That got me a bit further. But then some tests failed with:

```python
>       array = numpy.reshape(poly.values, shape=shape, newshape=newshape, order=order)
E       TypeError: reshape() got an unexpected keyword argument 'shape'
```

I also tried replacing `np.bool` with `bool`. That leads to the same errors when running tests.

**Comment by jonathf** [comment]
I've updated updated `np.bool -> np.bool_` in the Cython code and released version `1.3.5`.

@penguinpee, have you tried updating Numpy to version 2? I believe that is a braking change you are looking at.

**Comment by penguinpee** [comment]
Indeed, using NumPy 2.x (I tested locally in a venv with 2.1.3) makes the issue go away. However in that case you should adapt the following line accordingly.

https://github.com/jonathf/numpoly/blob/bad7759511b036d1a3fc674105830aae7d916242/pyproject.toml#L21

There is some ongoing effort in Fedora to update NumPy to 2.x. But that is still ongoing and will take some time to complete.

**Comment by jonathf** [comment]
Good point. Fixed in version `1.3.6`.
